pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -313,6 +324,30 @@ func (dsw *desiredStateOfWorld) VolumeExists(
 	return false
 }
 
+// DetermineVolumeAction determines action to be taken for pod volume
+// if the volume should be added or removed from desired state of world.
+// return values:
+//    true  -- if volume needs to be attached
+//    false -- if volume needs to be detached
+func (dsw *desiredStateOfWorld) DetermineVolumeAction(pod *v1.Pod, defaultValue bool) bool {
```

Comment:
This method is called in three places.
* `desired_state_of_world_populator.go:findAndRemoveDeletedPods()`
  * In this case the pods being operated on are fetched via a `dswp.desiredStateOfWorld.GetPodToAdd()` call so you can assume that the node is managed--no need to check again, just call `volumehelper.IsPodTerminated()`.
* `attach_detach_controller.go:podAdd()` and `attach_detach_controller.podUpdate()`
  * Both call this method first and then `util.ProcessPodVolumes(...)`.
  * `util.ProcessPodVolumes(...)` already calls `desiredStateOfWorld.NodeExists(...)` to verify the node is managed. So you don't need to do the check, `volumehelper.IsPodTerminated(...)` should be sufficient to determine if the volume should be added or deleted.

Additionally I would change this block of code:

```GO
if volumehelper.IsPodTerminated(pod, pod.Status) {
	util.ProcessPodVolumes(pod, false, /* addVolumes */
		adc.desiredStateOfWorld, &adc.volumePluginMgr, adc.pvcLister, adc.pvLister)
} else {
	util.ProcessPodVolumes(pod, true, /* addVolumes */
		adc.desiredStateOfWorld, &adc.volumePluginMgr, adc.pvcLister, adc.pvLister)
}
```
to 
```GO
addVolumes := !volumehelper.IsPodTerminated(pod, pod.Status)

util.ProcessPodVolumes(pod, addVolumes, adc.desiredStateOfWorld, &adc.volumePluginMgr, adc.pvcLister, adc.pvLister)
```

That will allow you to get rid of `desired_state_of_world.go:DetermineVolumeAction(...)` altogether.